packages/compiler/lib/decorators.ts

```diff
@@ -546,3 +516,32 @@ export function $friendlyName(
 export function getFriendlyName(program: Program, target: Type): string {
   return program.stateMap(friendlyNamesKey).get(target);
 }
+
+const knownValuesKey = Symbol();
+/**
+ * Specify the known values for a string type.
+ * @param target Decorator target. Must be a string. (model Foo extends string)
+ * @param knownValues Must be an enum.
+ */
+export function $knownValues(context: DecoratorContext, target: Type, knownValues: Type) {
+  if (
+    !validateDecoratorTargetIntrinsic(context.program, target, "@knownValues", [
+      "string",
+      "int8",
+      "int16",
+      "int32",
+      "int64",
+      "float32",
+      "float64",
+    ]) ||
+    !validateDecoratorParamType(context.program, target, knownValues, "Enum")
```

Comment:
Can we do a deeper evaluation here, to ensure the base type of the enum options match the intrinsic type of the target?

Reply:
added that validation, this becomes closer and closer to needing having a full type assignment checker with subtype/supertype relations